musicbot/bot.py

```diff
@@ -4429,82 +4449,53 @@ async def gen_cmd_list(self, message, list_all_cmds=False):
-            del timers[voice_channel.id]
+            except Exception as e:
+                log.info(
+                    f"Leaving voice channel {voice_channel.name} in {voice_channel.guild} due to inactivity."
+                )
+            await self.disconnect_voice_client(guild)
 
     async def on_voice_state_update(self, member, before, after):
         if not self.init_ok:
             return  # Ignore stuff before ready
 
         if self.config.leave_inactives:
-            guild_id = member.guild.id
-
-            # Ensure timers are initialized for this guild
-            if guild_id not in self.server_specific_data:
-                self.server_specific_data[guild_id] = {"voice_channel_timers": {}}
+            guild = member.guild
 
-            timers = self.server_specific_data[guild_id]["voice_channel_timers"]
+            event = self.server_specific_data[guild]["timeout_event"]
 
-            if before.channel and member != self.user:
+            if (
+                before.channel
+                and member != self.user
+                and self.user in before.channel.members
+            ):
                 if not any(not user.bot for user in before.channel.members):
-                    if not any(
-                        str(before.channel.id) in str(channel_id)
-                        for channel_id in [timers, self.config.autojoin_channels]
-                    ):
-                        timers[before.channel.id] = self.loop.create_task(
-                            asyncio.sleep(self.config.leave_inactiveVCTimeOut)
-                        )
-                        timers[before.channel.id].add_done_callback(
-                            lambda task: asyncio.ensure_future(
-                                self.on_timer_expired(before.channel)
-                            )
-                        )
-                        log.info(
-                            f"Started timer for inactive channel {before.channel.name} in {before.channel.guild}"
-                        )
-                    if str(before.channel.id) in str(self.config.autojoin_channels):
-                        log.info(
-                            f"Ignoring {before.channel.name} in {before.channel.guild} as it is a binded voice channel."
-                        )
-            elif after.channel:
-                if after.channel.id in timers:
-                    timers[after.channel.id].cancel()
-                    log.info(
-                        f"Cancelling timer for {after.channel.name} in {after.channel.guild} as channel is no longer inactive."
+                    log.debug("Channel is empty, should be handling disconnects now")
+                    await self.handle_timeout(guild)
+            elif after.channel and member != self.user:
+                if self.user in after.channel.members:
+                    log.debug(
+                        "Someone joined the channel again, should cancel the timer for disconnects"
```

Comment:
log.info(f"A user joined {after.channel.name}, cancelling timer."} 
I do like the rewording here though.